molbe/solver.py

```diff
@@ -1,13 +1,19 @@
-# Author(s): Oinam Romesh Meitei, Leah Weisburn
-
-import numpy,functools,sys, time,os
+# Author(s): Oinam Romesh Meitei, Leah Weisburn, Shaun Weatherly
+
+import numpy
+import functools
+import sys
+import time
+import os
+from molbe import be_var
 from molbe.external.ccsd_rdm import make_rdm1_ccsd_t1, make_rdm2_urlx, make_rdm1_uccsd, make_rdm2_uccsd
 
 def be_func(pot, Fobjs, Nocc, solver, enuc, hf_veff=None,
-            only_chem = False, nproc=4,hci_pt=False,
+            only_chem = False, nproc=4, hci_pt=False,
             hci_cutoff=0.001, ci_coeff_cutoff = None, select_cutoff=None,
             eeval=False, ereturn=False, frag_energy=False, relax_density=False,
-            return_vec=False, ecore=0., ebe_hf=0., be_iter=None, use_cumulant=True):
+            return_vec=False, ecore=0., ebe_hf=0., be_iter=None, use_cumulant=True, 
+            scratch=None, **solver_kwargs):
```

Comment:
Can this be modified so that it uses the existing `scratch_dir` attribute? [https://github.com/oimeitei/quemb/blob/0f3c6c2d7eea70d4a6fb2bdb4c292b8d7bfe8068/molbe/mbe.py#L167-L182](https://github.com/oimeitei/quemb/blob/0f3c6c2d7eea70d4a6fb2bdb4c292b8d7bfe8068/molbe/mbe.py#L167-L182)

Reply:
I've changed it to use this `scratch_dir` attribute by default and changed the variable names to reflect that. I had an issue of wanting to keep track of the individual fragment ins/outs to block2, so the original `scratch` attribute was easy to use as a parent directory for that.

Reply:
oh, i meant something slightly different -- i thought it would be useful to do a logic where
1. if `BE.scratch_dir` was set, use this in the solver
2. otherwise, use the input keyword
but i think we can address this in a later pr (**reminder to create an issue upon merge**) as this is okay as is